EndlessClient/Dialogs/TextMultiInputDialog.cs

```diff
@@ -154,9 +201,68 @@ protected override void OnDrawControl(GameTime gameTime)
             _spriteBatch.Begin();
             _spriteBatch.Draw(_background, DrawAreaWithParentOffset, _backgroundSourceRectangle, Color.White);
             _spriteBatch.Draw(_background, DrawPositionWithParentOffset + _bottomOverlayDrawPosition, _bottomOverlaySource, Color.White);
+            _spriteBatch.Draw(_background, DrawPositionWithParentOffset + _rightOverlayDrawPosition, _rightOverlaySource, Color.White);
+
+            _spriteBatch.Draw(_background, DrawPositionWithParentOffset + _rightOverlayDrawPosition, _rightOverlaySource, Color.White);
+
             _spriteBatch.End();
 
             base.OnDrawControl(gameTime);
         }
+
+        // OnUpdateControl needs to solely run if Dialog.Nine is in use
+
+        protected override void OnUpdateControl(GameTime gameTime)
+        {
+            if (_dialogSize != DialogSize.NineWithScroll)
+            {
+                base.OnUpdateControl(gameTime);
+                return;
+            }
+            _scrollBar.UpdateDimensions(9);
+
+            if (_inputBoxes.Length > _scrollBar.LinesToRender)
+            {
+                for (int i = 0; i < _inputBoxes.Length; i++)
```

Comment:
This approach looks correct but I don't like that this `OnUpdateControl` is executed unconditionally in its entirety every time (at least for `NineWithScroll` size).

What I'd do instead is track the previous scroll offset, and only execute this code if the previous scroll offset has changed. You'll see this pattern in just about every spot that uses a scroll bar in the source code - it reduces the amount of code executed from *all the code* to just the single condition.